app/main.py

```diff
@@ -1,8 +1,29 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person_info in people:
+        name = person_info["name"]
+        age = person_info["age"]
+        person = Person(name, age)
+        if "wife" in person_info and person_info["wife"] is not None:
+            wife_name = person_info["wife"]
+            wife_person = Person.people.get(wife_name)
```

Comment:
why do u need create var for name, if u are using it only once ?
u can get wife name directly in 18 line